src/main/java/bio/terra/common/UriUtils.java

```diff
@@ -0,0 +1,70 @@
+package bio.terra.common;
+
+import autovalue.shaded.kotlin.text.Charsets;
+import bio.terra.service.filedata.exception.InvalidUriException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.List;
+import java.util.Objects;
+import org.apache.http.NameValuePair;
+import org.apache.http.client.utils.URIBuilder;
+import org.apache.http.client.utils.URLEncodedUtils;
+
+/** Handy methods for working with URIs */
+public class UriUtils {
+
+  private UriUtils() {}
```

Comment:
Ah, I figured out how to avoid needing a no-op constructor (if you'd like): annotating the class as a `@Component` allows you to skip it.

Reply:
I would recommend against using `@Component` as a way to express a utils class, since making it a component means that it will be instantiated as part of the application context.

If you want to avoid adding a no-op constructor (and declaring the class final, etc) I'd recommend using the enum pattern instead. In that case you define an enum with no members:
```
public enum UriUtils {
    ;

   [...]
```

And no other code or annotations are needed. This also works if you want to represent a singleton object; in that case you can declare a single enum member which will be the singleton instance.